packages/celer-web-app/src/ui/frames/DocFrame.tsx

```diff
@@ -93,6 +154,7 @@ export const DocFrame: React.FC<DocFrameProps> = ({docLines})=>{
 			line = docLines[lineNumber];
 			if (line.lineType === "DocLineText" || line.lineType === "DocLineTextWithIcon") {
 				setDocCurrentLine(lineNumber);
+				setCurrentLine(origLineNumber-1);
```

Comment:
What's the difference between DocCurrentLine and CurrentLine? Do we truly need both?

Reply:
DocCurrentLine is the line used for centering the map. CurrentLine is the line that is scrolled to. We need both because not every line in the document has a location on the map, so DocCurrentLine scrolls down until it hits a line that does have a location on the map. If they were on the same variable, the keyboard controls would behave erratically as the map recenters.

Reply:
I'm not sure if I'm following "If they were on the same variable, the keyboard controls would behave erratically as the map recenters." The keyboard control should always go up or down one line based on the current line that's scrolled to and centers the map accordingly, so "line that is scrolled to" and "line used for centering the map" should in theory be the same

Reply:
The map recentering goes down through the document lines until it reaches a line that actually has coordinates on the map. This is not always going to be the line that is currently scrolled to, so if the keyboard controls used the same variable, it could scroll down multiple lines when you press the down arrow key. I suppose that I could move the scrolling down logic into the map, then consolidate them into one variable. Would that be preferable?

Reply:
I see. I would change it so that `setDocCurrentLine` is set before the loop on line 153 so it becomes the same as CurrentLine and we can consolidate that into one. Then the loop will search for the line to center to and calls `centerMapToLine`. I don't think we need to store which line is the line the map is centering too